test/test_torch.py

```diff
@@ -10427,7 +10429,8 @@ def check_norm(a, b, expected_norm, gels_result):
 
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
-    def test_qr(self, device):
+    @tf32_on_and_off(atol_=0.001, rtol_=0.001)
+    def test_qr(self, device, rtol, atol):
```

Comment:
wait, why is this being tested with tf32 on and off? I thought qr calls into magma, and magma is always true fp32?

Reply:
Becasue inside the test, it uses `matmul` to verify the result.

Reply:
If it uses matmul to verify result only, then maybe matmul should be run in fp32 in this case, and no changed tolerance?